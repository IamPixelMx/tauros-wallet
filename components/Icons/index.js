import CoffeeIcon from './coffee.svg';
import DessertIcon from './dessert.svg';
import LikeIcon from './like.svg';
import MelpIcon from './melp.svg';
import SaladIcon from './salad.svg';
import SearchIcon from './search.svg';
import ShareCircleIcon from './share-circle.svg';
import ShareIcon from './share.svg';
import OrientalIcon from './oriental.svg';

export {
  CoffeeIcon,
  DessertIcon,
  LikeIcon,
  MelpIcon,
  SearchIcon,
  SaladIcon,
  ShareCircleIcon,
  ShareIcon,
  OrientalIcon,
};
