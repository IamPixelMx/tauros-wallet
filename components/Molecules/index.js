import { Modal, ModalTitle } from './Modal';
import { Navbar } from './Navbar';

export { Modal, ModalTitle, Navbar };
