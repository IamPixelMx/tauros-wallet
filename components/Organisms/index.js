import { Balance } from './Balance';
import { Dashboard } from './Dashboard';
import { Layout } from './Layout';
import { Trading } from './Trading';
import { TransfersList } from './TransfersList';

export { Balance, Dashboard, Layout, Trading, TransfersList };
