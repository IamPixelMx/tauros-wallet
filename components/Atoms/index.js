import { Button } from './Button';
import { Card } from './Card';
import { CardSubtitle, CardTitle } from './CardContent';
import { Loader } from './Loader';
import { TitleTab } from './TitleTab';

import { AddIcon, BtcIcon, CloseIcon, DownIcon, MxnIcon, TaurosIcon, UpIcon } from './Icons';

export {
  Button,
  Card,
  CardSubtitle,
  CardTitle,
  AddIcon,
  BtcIcon,
  CloseIcon,
  DownIcon,
  MxnIcon,
  TaurosIcon,
  UpIcon,
  Loader,
  TitleTab,
};
