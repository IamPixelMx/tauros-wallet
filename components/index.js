import {
  Button,
  Card,
  CardSubtitle,
  CardTitle,
  AddIcon,
  BtcIcon,
  CloseIcon,
  DownIcon,
  MxnIcon,
  TaurosIcon,
  TaurosDarkIcon,
  UpIcon,
  Loader,
  TitleTab,
} from './Atoms';

import { Modal, ModalTitle, Navbar } from './Molecules';

import { Dashboard, Layout } from './Organisms';

export {
  Button,
  Card,
  CardSubtitle,
  CardTitle,
  Dashboard,
  AddIcon,
  BtcIcon,
  CloseIcon,
  DownIcon,
  MxnIcon,
  TaurosIcon,
  TaurosDarkIcon,
  UpIcon,
  Layout,
  Loader,
  Modal,
  ModalTitle,
  Navbar,
  TitleTab,
};
