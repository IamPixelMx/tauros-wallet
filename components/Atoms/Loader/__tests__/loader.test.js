import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import { Loader } from 'components';

expect.addSnapshotSerializer(serializer);

describe('Loader', () => {
  const loaderDefaultElement = <Loader />;

  const loaderFullScreenElement = <Loader isFullScreen />;

  const loaderCustomHeightElement = <Loader customHeight='500px' />;

  it('Renders default Loader', () => {
    const loader = renderer.create(loaderDefaultElement).toJSON();
    expect(loader).toMatchSnapshot();
  });

  it('Renders full screen Loader', () => {
    const loader = renderer.create(loaderFullScreenElement).toJSON();
    expect(loader).toMatchSnapshot();
  });

  it('Renders full screen with custom height', () => {
    const loader = renderer.create(loaderCustomHeightElement).toJSON();
    expect(loader).toMatchSnapshot();
  });
});
