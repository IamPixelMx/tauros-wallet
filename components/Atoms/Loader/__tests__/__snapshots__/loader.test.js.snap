// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Renders default Loader 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotateY(0deg);
    -ms-transform: rotateY(0deg);
    transform: rotateY(0deg);
  }

  50% {
    -webkit-transform: rotateY(180deg);
    -ms-transform: rotateY(180deg);
    transform: rotateY(180deg);
  }

  100% {
    -webkit-transform: rotateY(0deg);
    -ms-transform: rotateY(0deg);
    transform: rotateY(0deg);
  }
}

.emotion-6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 5rem;
  max-width: 100vw;
  width: 100%;
  background-color: var(--black-76);
  padding: var(--spacing-l) 0;
}

.emotion-4 {
  position: relative;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-animation-duration: 1500ms;
  animation-duration: 1500ms;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.emotion-0 {
  position: absolute;
  top: 50%;
  left: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 5rem;
  height: 5rem;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.emotion-0:first-of-type {
  z-index: 2;
  -webkit-transform: translate(-50%,-50%) rotateY(0deg);
  -ms-transform: translate(-50%,-50%) rotateY(0deg);
  transform: translate(-50%,-50%) rotateY(0deg);
}

.emotion-0:last-of-type {
  -webkit-transform: translate(-50%,-50%) rotateY(180deg);
  -ms-transform: translate(-50%,-50%) rotateY(180deg);
  transform: translate(-50%,-50%) rotateY(180deg);
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-0 emotion-1"
    >
      <span
        height="5rem"
        width="5rem"
      />
    </div>
    <div
      className="emotion-0 emotion-1"
    >
      <span
        height="5rem"
        width="5rem"
      />
    </div>
  </div>
</div>
`;

exports[`Loader Renders full screen Loader 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotateY(0deg);
    -ms-transform: rotateY(0deg);
    transform: rotateY(0deg);
  }

  50% {
    -webkit-transform: rotateY(180deg);
    -ms-transform: rotateY(180deg);
    transform: rotateY(180deg);
  }

  100% {
    -webkit-transform: rotateY(0deg);
    -ms-transform: rotateY(0deg);
    transform: rotateY(0deg);
  }
}

.emotion-4 {
  position: relative;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-animation-duration: 1500ms;
  animation-duration: 1500ms;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.emotion-0 {
  position: absolute;
  top: 50%;
  left: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 5rem;
  height: 5rem;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.emotion-0:first-of-type {
  z-index: 2;
  -webkit-transform: translate(-50%,-50%) rotateY(0deg);
  -ms-transform: translate(-50%,-50%) rotateY(0deg);
  transform: translate(-50%,-50%) rotateY(0deg);
}

.emotion-0:last-of-type {
  -webkit-transform: translate(-50%,-50%) rotateY(180deg);
  -ms-transform: translate(-50%,-50%) rotateY(180deg);
  transform: translate(-50%,-50%) rotateY(180deg);
}

.emotion-6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 5rem;
  max-width: 100vw;
  width: 100%;
  background-color: var(--black-76);
  padding: var(--spacing-l) 0;
  height: calc(100vh - (2 * var(--space-large) + var(--space-x-large)));
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-0 emotion-1"
    >
      <span
        height="5rem"
        width="5rem"
      />
    </div>
    <div
      className="emotion-0 emotion-1"
    >
      <span
        height="5rem"
        width="5rem"
      />
    </div>
  </div>
</div>
`;

exports[`Loader Renders full screen with custom height 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotateY(0deg);
    -ms-transform: rotateY(0deg);
    transform: rotateY(0deg);
  }

  50% {
    -webkit-transform: rotateY(180deg);
    -ms-transform: rotateY(180deg);
    transform: rotateY(180deg);
  }

  100% {
    -webkit-transform: rotateY(0deg);
    -ms-transform: rotateY(0deg);
    transform: rotateY(0deg);
  }
}

.emotion-4 {
  position: relative;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-animation-duration: 1500ms;
  animation-duration: 1500ms;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.emotion-0 {
  position: absolute;
  top: 50%;
  left: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 5rem;
  height: 5rem;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.emotion-0:first-of-type {
  z-index: 2;
  -webkit-transform: translate(-50%,-50%) rotateY(0deg);
  -ms-transform: translate(-50%,-50%) rotateY(0deg);
  transform: translate(-50%,-50%) rotateY(0deg);
}

.emotion-0:last-of-type {
  -webkit-transform: translate(-50%,-50%) rotateY(180deg);
  -ms-transform: translate(-50%,-50%) rotateY(180deg);
  transform: translate(-50%,-50%) rotateY(180deg);
}

.emotion-6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 5rem;
  max-width: 100vw;
  width: 100%;
  background-color: var(--black-76);
  padding: var(--spacing-l) 0;
  height: calc(500px - (2 * var(--space-large)));
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-0 emotion-1"
    >
      <span
        height="5rem"
        width="5rem"
      />
    </div>
    <div
      className="emotion-0 emotion-1"
    >
      <span
        height="5rem"
        width="5rem"
      />
    </div>
  </div>
</div>
`;
